### ISSUE TITLE
Add hint for internet permissions in readme

### ISSUE THREAD

**Issue by paulHempel** [nextcloud/neon#2123]
When using this package on MacOS or Android, one might get the error:
> [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ClientException with SocketException: Connection failed (OS Error: Operation not permitted, errno = 1), address = localhost, port = 8080, uri=...

Which is caused by not setting the permissions for internet access on certain platforms.

> If you are deploying to Android, edit your AndroidManifest.xml file to add the Internet permission.
> 
> ```
> 
> <uses-permission android:name="android.permission.INTERNET" />
> 
> ```
> Likewise, if you are deploying to macOS, edit your macos/Runner/DebugProfile.entitlements and macos/Runner/Release.entitlements files to include the network client entitlement.
> 
> ```
> 
> <key>com.apple.security.network.client</key>
> <true/>
> ```
https://docs.flutter.dev/cookbook/networking/fetch-data

I would suggest to add this in the readme; and if you are interested, I can create the corresponding PR

**Comment by provokateurin** [comment]
Hi, so far we do not support macOS. Which package are you trying to use and in what context?

**Comment by paulHempel** [comment]
Hi. Yes, I know that macOS not a supported platform – for your apps :)

What I did was using in my `main()` method
```
final client = NextcloudClient(
    Uri.parse('http://localhost:8080'),
    loginName: 'admin',
    appPassword: '4PSg6-fJZ79-pMXk6-cgpRk-NbE9a',
  );

  final response = await client.provisioningApi.users.getCurrentUser();
  print(response.body.ocs.data.id);
```
just to see if it works. I'm intending/thinking about using some parts of the REST API provided by this package in general.


This issue is more of an hint on the issue as I just encountered it and wanting to prevent the rest from being confused by the same error when not setting up the permissions for the platforms.

**Comment by paulHempel** [comment]
And I should mention, that I plan to use the Deck API in my app. Which I thought, could be contributed to this project, instead of only building it for my app.

**Comment by provokateurin** [comment]
> This issue is more of an hint on the issue as I just encountered it and wanting to prevent the rest from being confused by the same error when not setting up the permissions for the platforms.

From my perspective it is clear that a package that communications with a Nextcloud instance over the network needs that permission. I don't think any other pure dart package contains such a notice.

> And I should mention, that I plan to use the Deck API in my app. Which I thought, could be contributed to this project, instead of only building it for my app.

Cool! Please be aware that we will only accept new OpenAPI specs generated with https://github.com/nextcloud/openapi-extractor from the source code. Unfortunately https://github.com/nextcloud/deck is not able to do that yet, but I know the maintainer is interested in it. That would be something that needs to be done first.

May I ask you what kind of app you are building? Are you building a pure Deck app or is Deck only some part of it? If you are just building a Deck app I'd suggest to work with neon_framework which is exactly made for this purpose and we also have plans to implement Deck at some point: https://github.com/nextcloud/neon/issues/944

**Comment by paulHempel** [comment]
> From my perspective it is clear that a package that communications with a Nextcloud instance over the network needs that permission. I don't think any other pure dart package contains such a notice.

> Note
> Flutter applications may require [additional configuration](https://docs.flutter.dev/data-and-backend/networking#platform-notes) to make HTTP requests.
https://pub.dev/packages/http

I understand where you are coming from, but to me this is more of an requirement (like a needing a certain minimum flutter version), which should be explicitly stated.
In the case of a lib providing an http client, which may be the only thing needing internet permission in an app because it makes it unnecessary to use anything else, I would say that such a hint would be quite helpful for those without the required knowledge to spot this exact error.

---

https://github.com/nextcloud/neon/issues/944

Basically a Deck-only app. Good point on joining your efforts. I'll hop on the matrix and dig deeper in this repo.

**Comment by provokateurin** [comment]
Ok that note from the http package seems reasonable to me, could you make a PR to add the same note to the nextcloud package?

**Comment by Leptopoda** [comment]
> Hi. Yes, I know that macOS not a supported platform – for your apps :)

Just to clarify for future readers:
While we do not yet support iOS,macOS or windows with the neon_framework (PRs welcome) the nextcloud package should support all dart platforms.
Do not hesitate to report any bugs (for whatever package or platform).
